libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioAdapter.cs

```diff
@@ -87,71 +88,46 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
         }
 
         /// <summary>
-        /// Accept an incoming webhook request and convert it into a TurnContext which can be processed by the bot's logic.
+        /// Accept an incoming webhook httpRequest and convert it into a TurnContext which can be processed by the bot's logic.
         /// </summary>
-        /// <param name="request">A request object from Restify or Express.</param>
-        /// <param name="response">A response object from Restify or Express.</param>
+        /// <param name="httpRequest">A httpRequest object from Restify or Express.</param>
+        /// <param name="httpResponse">A httpResponse object from Restify or Express.</param>
         /// <param name="bot">A bot with logic function in the form `async(context) => { ... }`.</param>
         /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot bot, CancellationToken cancellationToken = default)
+        public async Task ProcessAsync(HttpRequest httpRequest, HttpResponse httpResponse, IBot bot, CancellationToken cancellationToken = default)
         {
-            response.StatusCode = 200;
-            await response.WriteAsync(string.Empty, cancellationToken).ConfigureAwait(false);
-
-            var twilioSignature = request.Headers["x-twilio-signature"];
-
-            var validationUrl = _options.ValidationUrl ?? (request.Headers["x-forwarded-proto"][0] ?? request.Protocol + "://" + request.Host + request.Path);
+            if (httpRequest == null)
+            {
+                throw new ArgumentNullException(nameof(httpRequest));
+            }
 
-            var requestValidator = new RequestValidator(_options.AuthToken);
+            if (httpResponse == null)
+            {
+                throw new ArgumentNullException(nameof(httpResponse));
+            }
 
-            using (var bodyStream = new StreamReader(request.Body))
+            if (bot == null)
             {
-                var payload = bodyStream.ReadToEnd();
+                throw new ArgumentNullException(nameof(bot));
+            }
 
-                var formattedPayload = payload.Replace("+", "%20");
+            var activity = ReadRequest(httpRequest);
 
-                var body = QueryStringToDictionary(formattedPayload);
+            // create a conversation reference
+            using (var context = new TurnContext(this, activity))
+            {
+                context.TurnState.Add("httpStatus", "200");
```

Comment:
Could we replace where the hardcoded "200" by the corresponding OK code in **HttpStatusCode**?